databallpy/utils/synchronise_tracking_and_event_data.py

```diff
@@ -272,6 +275,11 @@ def _needleman_wunsch(
             raise ValueError(
                 "An event was left unassigned, check your gap penalty values"
             )
+        else:
+            raise ValueError(
+                f"The algorithm got stuck due to an unexpected "
+                f"value of P[{i}, {j}]: {P[i, j]}"
+            )
```

Comment:
Its good to have a general error raised when something is not working as expected, however, for users of the package this is error will basically means that they can not do what they want to do. If we find instances that this error is raised, its good practice to find out why it does not work and capture this case somewhere else in the code. So the question is, can we reproduce the error, open another issue and than try if we can capture that specific case so it stays more applicable for users.

Reply:
I definitely agree that this is not the solution, but it should exist as a fallback, otherwise there is the risk of users getting stuck in the while with no indication of what is happening (just like I experienced and had to debug).

Reply:
Agreed, we'll definately keep it in. But do you also know the underlying problem why got in the infinite loop? Is this also in the Utrecht - RKC match or something else?

Reply:
Yes, the underlying problem is the similarity matrix ending up with all its elements `nan`, as I explained in #88 
This was indeed also about Utrecht - RKC.